src/components/HomePage/SearchResultsArea/Offer/OfferApplyButton.spec.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import OfferApplyButton from "./OfferApplyButton";
+import { Button, createTheme, DialogTitle } from "@material-ui/core";
+import { renderWithTheme } from "../../../../test-utils";
+import { act, fireEvent } from "@testing-library/react";
+
+describe("OfferApplyButton", () => {
+    const theme = createTheme();
+
+    describe("render", () => {
+        it("should render Button component", () => {
+            expect(
+                shallow(<OfferApplyButton />)
+                    .find(Button).exists()
+            ).toBe(true);
+        });
+        it("should render 'Apply' text", () => {
+            const wrapper = renderWithTheme(
+                <OfferApplyButton />,
+                { theme }
+            );
+
+            expect(wrapper.getByText("Apply"));
+        });
+    });
+
+    describe("OfferApplyDialog", () => {
+        // TODO O Dialog esta la na mesma
+        it("dialog should be closed", () => {
+            const wrapper = mount(<OfferApplyButton />);
+            expect(wrapper.find(DialogTitle).exists()).toBe(false);
+            wrapper.unmount();
```

Comment:
What does mount and unmount do? *curiosity*

Reply:
https://testing-library.com/docs/react-testing-library/api/#unmount

https://wasuradananjith.medium.com/testing-with-jest-and-enzyme-in-react-part-4-shallow-vs-mount-in-enzyme-d60cad73f85c

Links I could find @DoStini

Reply:
Without unmounting, the following tests were still applying on that DOM